public/app/features/plugins/sql/components/configuration/ConnectionLimits.tsx

```diff
@@ -1,27 +1,93 @@
 import React from 'react';
 
-import { FieldSet, InlineField } from '@grafana/ui';
+import { DataSourceSettings } from '@grafana/data';
+import { FieldSet, InlineField, InlineFieldRow, InlineSwitch } from '@grafana/ui';
 import { NumberInput } from 'app/core/components/OptionsUI/NumberInput';
 
-import { SQLConnectionLimits } from '../../types';
+import { SQLConnectionDefaults } from '../../constants';
+import { SQLConnectionLimits, SQLOptions } from '../../types';
 
 interface Props<T> {
-  onPropertyChanged: (property: keyof T, value?: number) => void;
+  onOptionsChange: Function;
+  options: DataSourceSettings<SQLOptions>;
   labelWidth: number;
-  jsonData: SQLConnectionLimits;
 }
 
 export const ConnectionLimits = <T extends SQLConnectionLimits>(props: Props<T>) => {
-  const { onPropertyChanged, labelWidth, jsonData } = props;
+  const { onOptionsChange, options, labelWidth } = props;
+  const jsonData = options.jsonData;
+  const autoIdle = jsonData.maxIdleConnsAuto !== undefined ? jsonData.maxIdleConnsAuto : false;
 
+  // Update JSON data with new values
+  const updateJsonData = (values: {}) => {
+    const newOpts = {
+      ...options,
+      jsonData: {
+        ...jsonData,
+        ...values,
+      },
+    };
+
+    return onOptionsChange(newOpts);
+  };
+
+  // For the case of idle connections and connection lifetime
+  // use a shared function to update respective properties
   const onJSONDataNumberChanged = (property: keyof SQLConnectionLimits) => {
     return (number?: number) => {
-      if (onPropertyChanged) {
-        onPropertyChanged(property, number);
-      }
+      updateJsonData({ property: number });
     };
   };
 
+  // Calculate the number of idle connections
+  // automatically based on maximum connection
+  // number. Currently the same.
+  const getAutoIdleConns = (number: number) => {
```

Comment:
just a nit but if idle conns = open conns do we need this function at all? Are we planning on changing the auto logic soon?

Reply:
Agreed here, it's too much now